File: scripts/potomo.js
/* Convert .po files to .mo files. */
/* eslint-disable no-console */

const fs = require('fs');
const path = require('path');
const helpers = require('./helpers');

const [, , rootPath = '.'] = process.argv;

const l10nPath = path.join(rootPath, 'resources', 'l10n');
fs.readdirSync(l10nPath).forEach(file => {
  if (fs.statSync(path.join(l10nPath, file)).isDirectory()) {
    try {
      const poPath = path.join(l10nPath, file, 'LC_MESSAGES', 'loot.po');
      const moPath = path.join(l10nPath, file, 'LC_MESSAGES', 'loot.mo');

      fs.accessSync(poPath, fs.R_OK);

      helpers.safeExecFileSync('msgfmt', [poPath, '-o', moPath]);
    } catch (error) {
      console.error(error);
    }
  }
});

File: scripts/po_to_mo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import subprocess

def run_msgfmt(locale_directory):
    po_path = os.path.join(locale_directory, 'LC_MESSAGES', 'loot.po')
    mo_path = os.path.join(locale_directory, 'LC_MESSAGES', 'loot.mo')

    if (os.path.exists(po_path)):
        subprocess.check_call(['msgfmt', po_path, '-o', mo_path])

if __name__ == "__main__":
    root_path = os.path.join('.', 'resources', 'l10n')

    locale_directories = [ f.path for f in os.scandir(root_path) if f.is_dir() ]
    for local_directory in list(locale_directories):
        run_msgfmt(local_directory)
